src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/RegressionTest.java

```diff
@@ -122,6 +127,109 @@ public void testSelectIntoUpdateCount() throws SQLException {
         if (null != con)
             con.close();
     }
+   
+    /**
+     * Tests update query
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testUpdateQuery() throws SQLException {
+        SQLServerConnection con = (SQLServerConnection) DriverManager.getConnection(connectionString);
+        String sql;
+        PreparedStatement pstmt = null;
+        JDBCType[] targets = {JDBCType.INTEGER, JDBCType.SMALLINT};
+        int rows = 3;       
+        final String tableName = "[updateQuery]";
+
+        Statement stmt = con.createStatement();
+        Utils.dropTableIfExists(tableName, stmt);
+        stmt.executeUpdate("CREATE TABLE " + tableName + " (" + "c1 int null," + "PK int NOT NULL PRIMARY KEY" + ")");
+
+        /*
+         * populate table
+         */
+        sql = "insert into " + tableName + " values(" + "?,?" + ")";
+        pstmt =  (connection).prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY,
+                ResultSet.CONCUR_READ_ONLY, connection.getHoldability());
+        
+        for (int i = 1; i <= rows; i++) {
+            pstmt.setObject(1, i, JDBCType.INTEGER);
```

Comment:
changing this to pstmt.setObject(1, i, java.sql.Types.INTEGER); should solve the issue for java 7.